Jenkinsfile

```diff
@@ -1,8 +1,65 @@
 #!/usr/bin/env groovy
 
-common {
-    slackChannel = '#tools-notifications'
-    downStreamValidate = false
-    runMergeCheck = false
-    nodeLabel = 'docker-oraclejdk8'
-}
+// Copyright 2020 Confluent Inc
+
+properties([
+        buildDiscarder(logRotator(artifactDaysToKeepStr: '', artifactNumToKeepStr: '', daysToKeepStr: '30', numToKeepStr: '')),
+        parameters([
+            string(name: 'RELEASE_TAG', defaultValue: '')
+        ])
+])
+
+// We use the oldest available jdk version.
+node('docker-oraclejdk8') {
+  stage('Source') {
+    checkout([
+        $class: 'GitSCM',
+        branches: scm.branches,
+        doGenerateSubmoduleConfigurations: scm.doGenerateSubmoduleConfigurations,
+        extensions: [[$class: 'CloneOption', noTags: false, shallow: false, depth: 0, reference: '']],
+        userRemoteConfigs: scm.userRemoteConfigs,
+    ])
+
+    // If we have a RELEASE_TAG specified as a build parameter, test that the version in pom.xml matches the tag.
+    if ( !params.RELEASE_TAG.trim().equals('') ) {
+        sh "git checkout ${params.RELEASE_TAG}"
+        def project_version = sh (
+                        script: 'mvn help:evaluate -Dexpression=project.version -q -DforceStdout | tail -1',
+                        returnStdout: true
+                        ).trim()
+
+        if ( !params.RELEASE_TAG.trim().equals(project_version) ){
+            echo 'ERROR: tag doesn\'t match project version, please correct and try again'
+            echo "Tag: ${params.RELEASE_TAG}"
+            echo "Project version: ${project_version}"
+            currentBuild.result = 'FAILURE'
+            return
+        }
+    }
+  }
+  stage('Build') {
+      archiveArtifacts artifacts: 'pom.xml'
+      withVaultEnv([["artifactory/tools_jenkins", "user", "TOOLS_ARTIFACTORY_USER"],
```

Comment:
why do you need to export all of these environment variables from vault? i don't think they are being used anywhere, except for gpg passphrase.